model/patch_lifecycle.go

```diff
@@ -659,136 +659,90 @@ func RetryCommitQueueItems(projectID string, patchType string, opts RestartOptio
 		}
 		return toBeRequeued, nil, nil
 	}
-	if patchType == commitqueue.SourcePullRequest {
-		restarted, notRestarted := restartPRItems(patches, cq)
-		return restarted, notRestarted, nil
-	}
-	if patchType == commitqueue.SourceCommandLine {
-		restarted, notRestarted := restartCLIItems(patches, cq)
-		return restarted, notRestarted, nil
-	}
-	return nil, nil, errors.Errorf("patch type '%s' is invalid", patchType)
-}
-
-func restartPRItems(patches []patch.Patch, cq *commitqueue.CommitQueue) ([]string, []string) {
-	restartedPatches := []string{}
-	patchesWithErrors := []string{}
+	patchesRestarted := []string{}
+	patchesFailed := []string{}
 	for _, p := range patches {
-		// reconstruct commit queue item from patch
-		modules := []commitqueue.Module{}
-		for _, modulePatch := range p.Patches {
-			if modulePatch.ModuleName != "" {
-				module := commitqueue.Module{
-					Module: modulePatch.ModuleName,
-					Issue:  modulePatch.PatchSet.Patch,
-				}
-				modules = append(modules, module)
-			}
-		}
-		item := commitqueue.CommitQueueItem{
-			Issue:   strconv.Itoa(p.GithubPatchData.PRNumber),
-			Modules: modules,
-			Source:  commitqueue.SourcePullRequest,
+		// use the PR number to determine if this is a PR or diff patch. Currently there
+		// is not a reliable field that is set for CLI patches
+		var err error
+		if p.GithubPatchData.PRNumber > 0 {
+			err = restartPRItem(p, cq)
+		} else {
+			err = restartCLIItem(p, cq)
 		}
-		if _, err := cq.Enqueue(item); err != nil {
+		if err != nil {
 			grip.Error(message.WrapError(err, message.Fields{
-				"patch":             p.Id,
-				"commit_queue":      cq.ProjectID,
-				"commit_queue_type": commitqueue.SourcePullRequest,
-				"item":              item.Issue,
-				"message":           "error enqueuing item",
-				"operation":         "restart failed commit queue versions",
+				"patch":        p.Id,
+				"commit_queue": cq.ProjectID,
+				"message":      "error restarting commit queue item",
```

Comment:
Can we also log p.GithubPatchData.PRNumber, so we don't have to look up the patch to know which function we sued?

Reply:
done